mozldap.go

```diff
@@ -389,32 +362,28 @@ func (cli *Client) GetUserSSHPublicKeys(shortdn string) (pubkeys []string, err e
 // or "uid=ffxbld". Do not add ,dc=mozilla to the DN.
 //
 // example: cli.GetUserPGPFingerprint("mail=jvehent@mozilla.com")
-func (cli *Client) GetUserPGPFingerprint(shortdn string) (fp string, err error) {
-	defer func() {
-		if e := recover(); e != nil {
-			err = fmt.Errorf("mozldap.GetUserPGPFingerprint(shortdn=%q) -> %v",
-				shortdn, e)
-		}
-	}()
+func (cli *Client) GetUserPGPFingerprint(shortdn string) (string, error) {
+	errorText := fmt.Sprintf("mozldap.GetUserPGPFingerprint(shortdn=%q)", shortdn)
+
+	fingerprint := ""
 	entries, err := cli.Search("", "("+shortdn+")", []string{"pgpFingerprint"})
 	if err != nil {
-		panic(err)
+		return "", fmt.Errorf("%s -> %q", errorText, err.Error())
 	}
-	for _, entry := range entries {
-		for _, attr := range entry.Attributes {
-			if attr.Name != "pgpFingerprint" {
-				continue
-			}
-			for _, fp = range attr.Values {
-				// remove spaces
-				fp = strings.Replace(fp, " ", "", -1)
-				if len(fp) == 40 {
-					return
-				}
-			}
-		}
+	if len(entries) != 1 {
+		return "", fmt.Errorf("%s -> found %d entries matching shortdn %q, expected 1", errorText, len(entries), shortdn)
 	}
-	panic("no fingerprint found in ldap")
+	if entries[0].Attributes[0] != nil && len(entries[0].Attributes[0].Values) == 1 {
```

Comment:
Same comment as previously about replacing the range logic.
Also, this looks like something that could be implemented in a small function on the `entries` type.